gridstatus/eia.py

```diff
@@ -149,6 +154,222 @@ def fetch_page_wrapper(url, headers, page, page_size):
 
         return df
 
+    def get_daily_spots_and_futures(self):
+        """
+        Retrieves daily spots and futures for select energy products.
+
+        Includes Wholesale Spot and Retail Petroleum, Natural Gas.
+        Prompt-Month Futures, broken on EIA side,
+        for Crude, Gasoline, Heating Oil, Natural Gas, Coal, Ethanol.
+
+        They are published daily and not persisted, so this should be run once daily.
+
+        Returns:
+            d: dictionary of DataFrames for each table of values."""
+
+        url = "https://www.eia.gov/todayinenergy/prices.php"
+
+        df_petrol = pd.DataFrame(columns=["product", "area", "price", "percent_change"])
+        df_ng = pd.DataFrame(
+            columns=[
+                "region",
+                "natural_gas_price",
+                "natural_gas_percent_change",
+                "electricity_price",
+                "electricity_percent_change",
+                "spark_spread",
+            ],
+        )
+
+        def contains_wholesale_petroleum(text):
+            return text and "Wholesale Spot Petroleum Prices" in text
+
+        with urlopen(url) as response:
```

Comment:
when we request a URL, let's make sure to call the log function that will print the request if `verbose=True`

for example: https://github.com/kmax12/gridstatus/blob/main/gridstatus/ercot.py#L1422